improver/precipitation_type/snow_splitter.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown copyright. The Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""module to separate snow and rain contributions from a precipitation diagnostic"""
+
+from typing import Tuple
+
+import numpy as np
+from iris import Constraint
+from iris.cube import Cube, CubeList
+
+from improver import BasePlugin
+from improver.cube_combiner import Combine
+from improver.utilities.cube_checker import assert_spatial_coords_match
+
+
+class SnowSplitter(BasePlugin):
+    """A class to separate the contribution of rain or snow from the precipitation
+    rate/accumulation. This is calculated using the probability of rain and snow
+    at the surface to determine what fraction of the precipitation rate/accumulation
+    is rain or snow.
+     """
+
+    def __init__(self, variable: str):
+        """
+        Sets up Class
+
+        Args:
+            variable:
+                A string of either rain or snow depending on which diagnostic's
+                rate/accumulation should be outputted
```

Comment:
As this can only have two acceptable values, can you change it to a bool? Something like `output_is_rain`? It removes the need for the exception if the developer capitalises the variable. Obviously, this requires a change in the CLI too.

Reply:
I've updated this to be a bool as suggested